mas-foundation/src/main/java/com/ca/mas/core/oauth/CodeVerifierCache.java

```diff
@@ -8,44 +8,45 @@
 
 package com.ca.mas.core.oauth;
 
+import com.ca.mas.core.storage.sharedstorage.SharedPreferencesUtil;
+
 /**
  * Temporary cache to store code verifier
  */
 public class CodeVerifierCache {
 
     private static CodeVerifierCache instance = new CodeVerifierCache();
-
-    private String state;
-    private String codeVerifier;
+    private SharedPreferencesUtil prefUtil = null;
 
     private CodeVerifierCache() {
+        prefUtil = new SharedPreferencesUtil("codeverifier");
     }
 
     public static CodeVerifierCache getInstance() {
         return instance;
     }
 
     public void store(String state, String codeVerifier) {
-        this.state = state;
-        this.codeVerifier = codeVerifier;
+        prefUtil.save("state", state);
+        prefUtil.save("code", codeVerifier);
```

Comment:
Is it a good idea to store code-verifier with state string as key ?

Reply:
Added the changes. state as key.